gateway/src/resty/oidc/discovery.lua

```diff
@@ -40,10 +46,41 @@ function _M.new(http_backend)
             ssl = { verify = resty_env.enabled('OPENSSL_VERIFY') }
         }
     }
+    return _M.new_with_http_client(http_client)
+end
+
+function _M.new_with_http_client(http_client)
+    local self = setmetatable({ http_client = http_client }, mt)
+    self:init_cache()
```

Comment:
Nitpick: I think that it's a bit unexpected that the method assigns `self.something`.
I'd change this to `self.cache = init_cache()` or just inline if it's simple enough.